docs/docs/guides/receiving/pull.md

```diff
@@ -6,5 +6,77 @@ parent: Receiving messages
 grand_parent: How to Guides
 ---
 
-Receiving messages using Pull
----
+# Receiving messages using Pull
+Metro supports both push and pull message delivery. For an overview and comparison of pull and push subscriptions, see the [Subscriber Overview](overview.md). This document describes pull delivery. For a discussion of push delivery, see the [Push Subscriber Guide](push.md).
+
+## Asynchronous Pull
+Using asynchronous pulling provides higher throughput in your application, by not requiring your application to block for new messages. Messages can be received in your application using a long running message listener, and acknowledged one message at a time, as shown in the example below. Java, Python, .NET, Go, and Ruby clients use the StreamingPull service API to implement the asynchronous client API efficiently.
```

Comment:
remove mention of client libs for now?

Reply:
we can use all these client libs with metro endpoint, right ?

Reply:
Need to solve for auth first 👀

Reply:
Basic auth implementation works for all clients ?

Reply:
Not basic auth. It expects auth key in the below format.

```
client, err := pubsub.NewClient(context.Background(), "your-project-id")
        if err != nil {
                return fmt.Errorf("pubsub.NewClient: %v", err)
        }
        defer client.Close()
        // Use the authenticated client.
        _ = client
```
 The client reads the authentication key from the below-mentioned env var.

Check this out https://cloud.google.com/docs/authentication
```
Download service account key per https://cloud.google.com/docs/authentication/production.
Set environment variable GOOGLE_APPLICATION_CREDENTIALS=/path/to/service-account-key.json
This environment variable will be automatically picked up by the client.
```

Reply:
Let me try a workaround with `option.WithAPIKey()` method

Reply:
Removing for nwo